### PR TITLE
`build_dsp` cleanup

### PR DESCRIPTION
**Pull request by gipert**
@iguinn could you check this?

### CODE REVIEW

File: src/pygama/dsp/processing_chain.py [hunk 1]
```diff
@@ -674,22 +682,24 @@ def get_index(slice_value):
                     var.update_auto(*args, **kwargs)
                     return self._vars_dict[var_name]
                 elif not dry_run:
-                    print(args, kwargs)
+                    log.debug(f"{args} {kwargs}")
```
Comment (iguinn) [inline src/pygama/dsp/processing_chain.py:685]: This line can be removed
```suggestion
```

File: src/pygama/dsp/build_dsp.py [hunk 2]
```diff
@@ -128,36 +130,37 @@ def build_dsp(f_raw, f_dsp, dsp_config, lh5_tables=None, database=None,
...
+        if n_max and n_max < tot_n_rows:
+            tot_n_rows = n_max
 
         # if we have separate DSP files for each table, read them in here
         if chan_config is not None:
             f_config = chan_config[tb]
             with open(f_config) as config_file:
-                dsp_config = json.load(config_file, object_pairs_hook=OrderedDict)
-            print('Processing table:', tb, 'with DSP config file:\n  ', f_config)
+                dsp_config = json.load(config_file, object_pairs_hook=json.OrderedDict)
```
Comment (iguinn) [inline src/pygama/dsp/build_dsp.py:140]: object_pairs_hook is not needed; order does not matter
```suggestion
                dsp_config = json.load(config_file)
```

File: src/pygama/dsp/build_dsp.py [hunk 3]
```diff
@@ -128,36 +130,37 @@ def build_dsp(f_raw, f_dsp, dsp_config, lh5_tables=None, database=None,
...
-        lh5_it = lh5.LH5Iterator(f_raw, tb, buffer_len = buffer_len)
+        lh5_it = lh5.LH5Iterator(f_raw, tb, buffer_len=buffer_len)
         proc_chain = None
         for lh5_in, start_row, n_rows in lh5_it:
             # Initialize
             if proc_chain is None:
-                proc_chain, lh5_it.field_mask, tb_out = build_processing_chain(lh5_in, dsp_config, db_dict, outputs, verbose, block_width)
-                tqdm_it = iter(tqdm_range(0, tot_n_rows, buffer_len, verbose))
+                proc_chain, lh5_it.field_mask, tb_out = build_processing_chain(lh5_in, dsp_config, db_dict, outputs, block_width)
+                progress_bar = tqdm(desc=f'Processing table {tb}', total=tot_n_rows,
```
Comment (iguinn) [inline src/pygama/dsp/build_dsp.py:162]: Should this be `if log.level <= logging.INFO` like the update below?
```suggestion
                if log.level <= logging.INFO:
                    progress_bar = tqdm(desc=f'Processing table {tb}', total=tot_n_rows,
```

File: src/pygama/dsp/build_dsp.py [hunk 4]
```diff
@@ -1,76 +1,81 @@
...
-        Number of waveforms to read/write from disk at a time. Default 3200.
-    block_width : int (optional)
-        Number of waveforms to process at a time. Default 16.
-    verbose : int (optional)
-        0: Silent except for exceptions thrown
-        1: Minimal information printed (default)
-        2: Verbose output for debugging purposes
-    chan_config
-        DOC_ME
+        name of JSON file containing
```
Comment (iguinn) [inline src/pygama/dsp/build_dsp.py:40]:
```suggestion
        dict or name of JSON file containing
```

File: src/pygama/dsp/build_dsp.py [hunk 5]
```diff
@@ -1,76 +1,81 @@
...
 
     Parameters
     ----------
     f_raw : str
-        Name of raw LH5 file to read from
+        name of raw LH5 file to read from
     f_dsp : str
-        Name of dsp LH5 file to write to
+        name of dsp LH5 file to write to
     dsp_config : str
```
Comment (iguinn) [inline src/pygama/dsp/build_dsp.py:39]:
```suggestion
    dsp_config : str or dict
```

**Review by iguinn — CHANGES_REQUESTED** [review]
Thanks for taking care of all of this!

**Comment by gipert** [comment]
@iguinn thanks a lot for reviewing! Can you help me now with type annotations? I'm sure I made mistakes...

**Review by iguinn — APPROVED** [review]

**Comment by iguinn** [comment]
I will test this before we merge refactor into main, and I'll also have a look at the type annotations

**Pull request merged**